eventually/static/src/containers/fileUpload/FileUpload.js

```diff
@@ -0,0 +1,152 @@
+            data.append('image', image);
+            axios.post(
+                request_path, 
+                data,
+                {timeout: request_timeout})
+                .then( (response)=>{
+                    if (response.status == 200) {
+                        this.setState({'image_name': response.data['image_key']});
+                        this.setState({'image_src': this.state.image_src.slice(0,53)+response.data['image_key']});
+                        this.hideLinearProgress();
+                        this.dialogClose();
+                        this.setState({'snackbarMessage': 'Image uploaded'});
+                        this.showSnackbar();}
+                    //Add else for other codes!
+                })
+                .catch( (error) => {
+                    this.setState({'snackbarMessage': 'Upload timed out'});
+                    this.hideLinearProgress();
+                    this.dialogClose();
+                    this.showSnackbar();
+                });
+        }
+        else {
+            this.hideLinearProgress();
+            this.setState({'snackbarMessage': 'Please choose another file'});
+            this.showSnackbar();
+        }
+    }
+
+    render() {
+        const actions = [
+            <FlatButton
+                label='Cancel'
+                primary={true}
+                keyboardFocused={false}
+                onClick={this.dialogClose}
+            />
+        ];
+
+        return (
+            <div>
+                <RaisedButton label='Upload image' primary={true} onClick={this.dialogOpen} />
+                <Dialog
+                    title='File Upload'
+                    actions={actions}
+                    modal={false}
+                    open={this.state.open}
+                    style={dialogStyle}
+                    onRequestClose={this.dialogClose}>
+                    <Dropzone
+                        multiple={false}
+                        style={dropzoneStyle}
+                        onDrop={this.onImageDrop}>
+                        <p>Drop an image here or click to upload the file.<br/>
+                        The file must be of GIF, PNG, JPG or BMP type.<br/>
+                        Maximum file size: 8 Mb</p>
+                    </Dropzone>
+                    <div style={{'position': 'relative', 'top':'20px'}}>
+                        <LinearProgress mode='indeterminate' style = {{'visibility': this.state.linearProgressVisibility}}/>
+                    </div>
+                </Dialog>
+                <Snackbar
+                    style={{'textAlign':'center'}}
+                    open={this.state.snackbarOpen}
+                    message={this.state.snackbarMessage}
+                    autoHideDuration={5000}
+                    onRequestClose={this.hideSnackbar}
+                />
+                {/* remove once not needed! */}
+                <div style={{   'margin': '10px 20px',
```

Comment:
move to variable

Reply:
This code isn't needed in the final version. Removed.